src/plugins/data/server/search/collectors/usage.ts

```diff
@@ -38,44 +40,53 @@ export interface SearchUsage {
   trackSuccess(duration: number): Promise<void>;
 }
 
-export function usageProvider(core: CoreSetup): SearchUsage {
+export function usageProvider(
+  core: CoreSetup,
+  initializerContext: PluginInitializerContext<ConfigSchema>
+): SearchUsage {
   const getTracker = (eventType: keyof Usage) => {
     return async (duration?: number) => {
-      const repository = await core
-        .getStartServices()
-        .then(([coreStart]) => coreStart.savedObjects.createInternalRepository());
+      const config = await initializerContext.config
+        .create<ConfigSchema>()
+        .pipe(first())
+        .toPromise();
```

Comment:
Since this is a one time read, why not generate the `config` in plugin `setup` function, and then simply pass the `config` object as the parameter?

Reply:
Yes. we tried it in the plugin setup function and got weird behavior, and setup was not async function in the search service plugin.

Reply:
Or maybe following what `autocomplete` service does at https://github.com/opensearch-project/OpenSearch-Dashboards/blob/f0ea74e8a2f4db70cfb66e2d5a68d57336c0eadf/src/plugins/data/server/autocomplete/autocomplete_service.ts#L36 to code in a more reactive way?

Reply:
Thanks for sharing the autocomplete sample, yes. we can refactor the search service later with more reactive way.